### PR TITLE
Inv subst

### PR DESCRIPTION
**Pull request by andreistefanescu**
fixes https://github.com/GaloisInc/saw-script/issues/1910

### CODE REVIEW

File: src/SAWScript/Crucible/LLVM/Builtins.hs [hunk 1]
```diff
@@ -1692,14 +1696,16 @@ verifyPoststate cc mspec env0 globals ret mdMap =
 
 -- | Translate the proof obligations from the Crucible backend into SAWCore
 -- terms. For each proof oblication, return a triple consisting of the error
--- message, the metadata, and the SAWCore.
+-- message, the metadata, and the SAWCore. For each proof obligation, substitute
+-- the uninterpreted invaiants with their definitions.
```
[the remark this answers is not in the document]
Comment (kquick) [reply src/SAWScript/Crucible/LLVM/Builtins.hs]: Yes, this is split out from 2016 to supply the plumbing that will actually be utilized in 2016.

**Review by kquick — APPROVED** [review]

**Comment by andreistefanescu** [comment]
@RyanGlScott as it turns out, you get a parting gift from me: fixing the CI here. I *think* this fails because of the simplification changes in `What4`. I assume fixing this this requires some tinkering with the existing rewrite rules. This need to be fixed regardless of the other open pull requests, as it prevents any further updates to the What4 submodule.

**Comment by RyanGlScott** [comment]
Indeed, I also suspect that the `what4` simplification rules introduced in https://github.com/GaloisInc/what4/pull/252 are responsible for the `awslc` CI job going into a (seemingly) infinite loop. Most likely, the proof goals now consist of different terms, which causes key rewrite rules not to fire. (See also https://github.com/GaloisInc/crucible/pull/1165, where bumping the `what4` submodule on the `crucible` side caused a What4 term in a test case to take on a very different shape.)

Judging from [the failing CI job](https://github.com/GaloisInc/saw-script/actions/runs/7645287816/job/20834335873?pr=2018#step:8:4511):

```
[21:06:03.395] Checking proof obligations EVP_DigestUpdate ...
Error: The operation was canceled.
```

The infinite loop occurs somewhere in the proof of `EVP_DigestUpdate`.

**Comment by RyanGlScott** [comment]
Superseded by #2037.

**Pull request closed without merging**